### PR TITLE
Update `@sentry/nextjs` to `7.16.0`

### PR DESCRIPTION
**Pull request by kachkaev**
Follows #682, relates to https://github.com/getsentry/sentry-javascript/issues/5998

### CODE REVIEW

File: apps/site/next.config.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const nextConfig = {
   swcMinify: true,
 
   sentry: {
+    autoInstrumentServerFunctions: true,
```
Comment (kachkaev) [inline apps/site/next.config.js:16]: https://github.com/getsentry/sentry-javascript/issues/5998#issuecomment-1285122737

File: apps/site/next.config.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const nextConfig = {
   swcMinify: true,
 
   sentry: {
+    autoInstrumentServerFunctions: false,
```
Comment (kachkaev) [inline apps/site/next.config.js:16]: https://github.com/getsentry/sentry-javascript/issues/5998#issuecomment-1285122737

File: apps/site/sentry.client.config.ts [hunk 3]
```diff
@@ -15,10 +15,7 @@ Sentry.init({
   enabled: !!dsn,
   environment: process.env.NEXT_PUBLIC_VERCEL_ENV ?? "unset",
   integrations:
-    replaysSamplingRate > 0 &&
-    replaysSamplingRate <= 1 &&
-    // @todo Remove when https://github.com/getsentry/sentry-replay/issues/246#issuecomment-1284472286 is resolved
-    typeof window !== "undefined"
+    replaysSamplingRate > 0 && replaysSamplingRate <= 1
```
[the remark this answers is not in the document]
Comment (kachkaev) [reply apps/site/sentry.client.config.ts]: [E2E tests failed](https://github.com/blockprotocol/blockprotocol/actions/runs/3291080901/jobs/5424917094), reverted

[the remark this answers is not in the document]
Comment (kachkaev) [reply apps/site/sentry.client.config.ts]: Tried using `@sentry/replay": "0.6.5` again in https://github.com/blockprotocol/blockprotocol/pull/694/commits/b8221dd942bc2c0e1d603402be5fc689086d76e2 to rule out an unlikely one-off Vercel issue. Still got failed CI. Reverted this change one more time 😅

[the remark this answers is not in the document]
Comment (kachkaev) [reply apps/site/sentry.client.config.ts]: From Vercel lambda logs:

Screenshot 2022-10-20 at 18 07 53

```
[GET] /api/logout
18:07:43:48
Function Status:404
Edge Status:500
Duration:150.14 ms
Init Duration: N/A
Memory Used:214 MB
ID:lhr1::iad1::ppzbd-1666285663041-8097e2919060
User Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36

[GET] /api/signup
18:07:41:78
Function Status:404
Edge Status:500
Duration:122.90 ms
Init Duration: N/A
Memory Used:213 MB
ID:lhr1::iad1::jnzg8-1666285661512-5c2386677c1a
User Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36

[GET] /api/me
18:07:37:52
Function Status:404
Edge Status:500
Duration:182.88 ms
Init Duration: N/A
Memory Used:213 MB
ID:lhr1::iad1::cvbkf-1666285655569-5623b05a3844
User Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/106.0.0.0 Safari/537.36
```

🤷

[the remark this answers is not in the document]
Comment (kachkaev) [reply apps/site/sentry.client.config.ts]: Upstream reply: https://github.com/getsentry/sentry-replay/issues/246#issuecomment-1285887129

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**